### PR TITLE
#165305756 Password Reset (Authentication)

### PR DESCRIPTION
**Pull request by Ogutu-Brian**
#### Description
Currently, when a user forgets his/her password, they are not able to make a request to reset it. This feature enables a user to get a password reset link sent to their emails on request to change their passwords

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [ ] End to end
- [x] Integration

#### Checklist:
- [x] My code follows the style guide for this project
- [x] I have linted my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[165305756](https://www.pivotaltracker.com/story/show/165305756)

### CODE REVIEW

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -73,3 +93,153 @@ def update(self, request, *args, **kwargs):
...
+            else:
+                try:
+                    user = User.objects.get(email=email)
+                except ObjectDoesNotExist:
+                    errors["email"] = ["no account with that email address"]
+                    response_status = status.HTTP_406_NOT_ACCEPTABLE
+                else:
+                    token, created = Token.objects.get_or_create(user=user)
+                    if not created:
+                        token.created = timezone.now()
```
Comment (IssaIan) [inline authors/apps/authentication/views.py]: @Ogutu-Brian you could implement the sending of email functions in one file, say mailer.py file, from where all email sending functions would be called. You and @Bakley could work on this file together. This keeps our views cleaner and implements good coding practices.

Comment (Ogutu-Brian) [reply authors/apps/authentication/views.py]: Let me do that @IssaIan

Comment (Ogutu-Brian) [reply authors/apps/authentication/views.py]: It is done

**Review by Amoswachira — APPROVED** [review]
LGTM

**Review by IssaIan — APPROVED** [review]
Good work Brian. 
LGTM

**Review by a-braham — APPROVED** [review]
LGTM

**Review by matthenge — APPROVED** [review]
LGTM

**Comment by mdn-network** [comment]
LGTM

**Review by mdn-network — APPROVED** [review]

**Review by mdn-network — APPROVED** [review]

**Pull request merged**